extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -44,17 +44,22 @@
             AuthenticationRequestContext context) {
         ContextAwareTokenCredential credential = (ContextAwareTokenCredential) request.getToken();
         RoutingContext vertxContext = credential.getContext();
-
-        if (tenantResolver.isBlocking(vertxContext)) {
-            return context.runBlocking(new Supplier<SecurityIdentity>() {
-                @Override
-                public SecurityIdentity get() {
-                    return authenticate(request, vertxContext).await().indefinitely();
+        return Uni.createFrom().deferred(new Supplier<Uni<SecurityIdentity>>() {
+            @Override
+            public Uni<SecurityIdentity> get() {
+                if (tenantResolver.isBlocking(vertxContext)) {
```

Comment:
Can it help with #8559 ?

Reply:
Possibly, I would need to investigate before I could say one way or the other.

Reply:
I suspect not, it is likely to do with the proactive autentication being off